---
title: "Class 12- Transcriptomics and the analysis of RNA-Seq data"
author: "rachel galleta (A16859649)"
format: pdf
toc: TRUE
---

## Background
Today we will analyze some RNASeq data from Himes et al. on the effects of a common steroid
(dexamethasone,) on airway smooth muscle cells (ASm cells)
Are staring point is the “counts” data and “metadata” that contain the count values for each
gene in their different experiments (i.e cell lines with or without drugs)

## Data import

```{r}
counts <- read.csv("airway_scaledcounts.csv", row.names=1)
metadata <-read.csv("airway_metadata.csv")
```

```{r}
head(counts)
```
Q1. How many genes are in this dataset? 

```{r}
nrow(counts)
```

Q2. How many ‘control’ cell lines do we have? 

```{r}
sum(metadata$dex=="control")
```

## Toy differential gene expression

To start our analysis lets calculate the mean for all genes in the "control" experiments.

1. extract all "control" columns form the counts objets 
2. Calculate the mean for all rows (ie, genes) of these "control"  columns 
3-4. Do the same for "treated"
5. Compare these "control.mean" and "treated.mean" values.

```{r}
#1.
control.inds <- metadata$dex=="control"
control.counts<- counts[,control.inds]
```


```{r}
#2.
control.means <- rowMeans( control.counts)
```

```{r}
dim(control.counts)
```


```{r}
#3-4.
treated.inds <- metadata$dex=="treated"
treated.counts<- counts[,treated.inds]
treated.means <- rowMeans(treated.counts)
```


store these together for ease of book means counts

```{r}
#5.
meancounts <- data.frame(control.means, treated.means)
head(meancounts)
```
mamke a plot onf ocntorl vs treatments menad valeues for all genes

```{r}
plot(meancounts)
```

Make this a log plot

```{r}
plot(meancounts,log="xy")
```

we often talk about metrics like "log2 fold-change"

```{r}
#treated/control
log2(10/10)
```

```{r}
log2(10/20)
```

```{r}
log2(20/10)
```

```{r}
log2(40/10)
log2(10/40)
```
lets calculate the log2 fold change four our treated over control mean counts

```{r}
meancounts$log2fc<-
log2(meancounts$treated.means/
  meancounts$control.means)
```

```{r}
head(meancounts)
```
A common "rule of thumb" is a log2 fold change cutoff of +2 and -2 to call genes "up regulated" or "down regulated"

Number of "up" genes
```{r}
sum(meancounts$log2fc >= +2, na.rm=T)
```

number of "down" genes at -2 threshold

```{r}
sum(meancounts$log2fc <= -2, na.rm=T)
```
##DESeq2 analysis

lest do  this analysis properlty and keep our inner starts nerd happy are the teh diferrenes we seen and no drug significnat givne the replciate experimnet

```{r,message=FALSE}
library(DESeq2)
```
for DESeq analysis we need thre things
-count values ('countData')
-metadat telling us about the columns in 'counData' ('colData')
-design of the experiemnt (what do you want to compare)

our first function formDESeq 2 will set up the input required for analysis by storing all these 3 thigs together.

```{r}
dds<- DESeqDataSetFromMatrix(countData=counts,
                             colData= metadata,
                             design= ~dex)
```

the main function in DESeq2  thats runs the nalysis is called DESeq()

```{r}
dds<- DESeq(dds)
```

```{r}
res<-results(dds)
head(res)
```

## Volcano Plot

this us common summary result fomr figrue thse tyoes of expreimnts and plot the log2 fold change  vs the adjusted p-value

```{r}
plot(res$log2FoldChange,-log(res$padj))
abline(v=c(-2,2),col="red")
abline(h=-log(0.04),col="red")
```


## save our results

```{r}
write.csv(res,file="my_results.csv")
```

## add gene anotation

to help us make sense the results and comunicate them to other folks we need to add some more annotation to our main `res` object

we will use two bioconductor packages to first map IDS to different formats including teh clasic gene "symbol" gene name.

`BiocManager::install("AnnotationDbi")`
`BiocManager::install("org.Hs.eg.db")

```{r}
library(AnnotationDbi)
library(org.Hs.eg.db)
```
let's see what is in `org.Hs.eg.db` with the `columns()` function:

```{r}
columns(org.Hs.eg.db)
```
We can translate or "map" IDs between any of these 26 database using the `mapIds()` function.

```{r}

res$symbol<-mapIds(keys=row.names (res), #our current file
       keytype = "ENSEMBL", #the format of our Ids 
       x= org.Hs.eg.db,         #Where to get the mapping from
       column = "SYMBOL")    # the format/DB to map to

head(res)
```
Add the mapping for "GENENAME" AND "ENTREZID" and store as `res$genenome` and `res$entrez`

```{r}
res$entrez<-mapIds(keys=row.names (res), 
       keytype = "ENSEMBL",
       x= org.Hs.eg.db,        
       column = "ENTREZID") 

head(res)

res$genename<-mapIds(keys=row.names (res), 
       keytype = "ENSEMBL",
       x= org.Hs.eg.db,        
       column = "GENENAME") 

head(res)
```

## Pathway Analysis 

there are lots of bioconductor packages to do this type of analysis. for now lets just try one called **gage** again we need to install thid if we dont have it already.

```{r message=FALSE}
library(gage)
library(gageData)
library(pathview)
```

To use **gage** I need two things 
- a name vector of fold- change values for ourDEGs (our geneset of interest)
- a set of pathways or genesets to use for annotation.

```{r}
x<-c(5,10)
```

```{r}
names(x)<-c("low","high")
x
```

```{r}
foldchanges<- res$log2FoldChange
names(foldchanges)<- res$entrez
head(foldchanges)
```

```{r}
data(kegg.sets.hs)

keggres = gage(foldchanges, gsets=kegg.sets.hs)
```

In our results object we have:
```{r}
attributes(keggres)
```
```{r}
head(keggres$less,5)
```

lets look at one of these pathways with our genes colored up so we can see the overlap.

```{r}
pathview(pathway.id = "hsa05310",gene.data=foldchanges)
```
Add this pathway figure to our lab report:

![](hsa05310.pathview.png)

## Save our main results 

```{r}
write.csv(res,file="myresults_annotated.csv")
```

