---
title: "class 11: structural bioinformatic pt1"
author: "rachel galleta"
format: html
---

## AlphaFold Data Base (AFBD)
the EBI maintains teh largest database of alphafold structure prediction models at: 

from last class (before halloween) we saw that the PDB had 244,290 (Oct 2025)

the total number of proteind sequences un UniprotKB is 199,579,901

>Key point:This is a tiny fraction of sequence space that has structural coverage (0.12%)

```{r}
244290/199579901*100
```

AFDB is attemping to address this gap..

there are two "Quality Scores" from teh alphafold one for residues (ie, each amino acid ) called pLDDT score. The other PAE score that measures the confidence in the relative position of two residues(i.e a socre for every )

## Generating your own structure predictions

figure of 5 generated HIV PR 

![](hvpr_dimer.png)
![]()









## Custom analysis of resulting models in R
Read Key result files into R, The first thing I nedd to know is what my reults directry /folder is called.?(ie )

```{r}
results_dir <-"HIVPR_dimer_23119/"
pdb_files <- list.files(path=results_dir,
                        pattern="*.pdb",
                        full.names = TRUE)

# Print our PDB file names
basename(pdb_files)

```



## Residue conservation from alignment file
```{r}
aln_file <- list.files(path=results_dir,
                       pattern=".a3m$",
                        full.names = TRUE)
aln_file
```






