---
title: "class 17- Introduction to Genome Informatics Lab"
author: "Rachel Galleta (A16859649)"
format: gfm
---

# section 1. Proportion on G/G in a population

downloaded a CSV file from Esemble < https://useast.ensembl.org/Homo_sapiens/Variation/Sample?db=core;r=17:39395098-40395098;v=rs8067378;vdb=variation;vf=959672880#373531_tablePanel

here we read this CSV file.

```{r}
mxl<- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```

```{r}
table(mxl$Genotype..forward.strand.)
```
```{r}
table(mxl$Genotype..forward.strand.) / nrow(mxl) * 100
```
Now lets look at a different population, I picked the GBR.


```{r}
gbr<- read.csv("373522-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")

```

find the prportion of G/G

```{r}
round (table(gbr$Genotype..forward.strand.) / nrow(gbr) * 100,2 )
```
this variant that is associated with childhood asthma is  more frequent in the GBR population than MKL population.

Lets now dig into this further.

## Section 4: Population Scale Analysis

One sample is obviously not enough to know what is happening in a population. You are interested in assessing genetic differences on a population scale. So, you processed about ~230 samples and did the normalization on a genome level. Now, you want to find whether there is any association of the 4 asthma-associated SNPs (rs8067378...) on ORMDL3 expression.

How many samples do we have?

```{r}
expr<- read.table("rs8067378_ENSG00000172057.6.txt")
head(expr)
```

```{r}
nrow(expr)
```
```{r}
table(expr$geno)
```
```{r}
library(ggplot2)
```
lets make a boxplot

```{r}
ggplot(expr)+ aes(geno, exp, fill=geno)+
  geom_boxplot(notch=TRUE)
```
what could you infer from the relative expression value between A/A and G/G displayed in this plot? Does the SNP effect the expression of ORMDL3?

A/A has the highest median expression, and G/G is the lowest. The SNP affects ORMDL3 expression because the expression difference between A and G is really different.
